### PR TITLE
Apex Application for benchmarking

### PR DESCRIPTION
**Pull request by sandeshh**
Without the Dimensional computation operator and HDHT integration

### CODE REVIEW

File: apex-benchmarks/XmlJavadocCommentsExtractor.xsl [hunk 1]
```diff
@@ -0,0 +1,46 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+
+    Copyright (C) 2015 DataTorrent, Inc.
```
Comment (brightchen) [inline apex-benchmarks/XmlJavadocCommentsExtractor.xsl]: Copyright should be 2016

File: apex-benchmarks/pom.xml [hunk 2]
```diff
@@ -0,0 +1,395 @@
...
+
+  <!-- change these to the appropriate values -->
+  <name>Apex_Benchmark</name>
+  <description>My Apex Application Description</description>
+
+  <properties>
+    <!-- change this if you desire to use a different version of DataTorrent -->
+    <datatorrent.version>3.4.0</datatorrent.version>
+    <datatorrent.apppackage.classpath>lib/*.jar</datatorrent.apppackage.classpath>
+    <megh.version>3.3.0</megh.version>
```
Comment (brightchen) [inline apex-benchmarks/pom.xml]: what megh version are you going to use? I think we agreed to use 3.4.0-SNAPSHOT

File: apex-benchmarks/pom.xml [hunk 3]
```diff
@@ -0,0 +1,395 @@
...
+    <malhar.version>3.4.0-incubating-SNAPSHOT</malhar.version>
```
Comment (brightchen) [inline apex-benchmarks/pom.xml]: should be 3.4.0-SNAPSHOT

File: apex-benchmarks/src/main/java/com/example/Application.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
+/**
+ * Licensed under the terms of the Apache License 2.0. Please see LICENSE file in the project root for terms.
```
Comment (brightchen) [inline apex-benchmarks/src/main/java/com/example/Application.java]: which license? DataTorrent or apache? should be consistent

File: apex-benchmarks/src/main/java/com/example/JsonGenerator.java [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+            for (String ad : entry.getValue()) {
+                ads.add(ad);
+            }
+        }
+        return ads;
+    }
+
+    @Override
+    public void emitTuples() {
+        out.emit( generateElement() ) ;
```
Comment (brightchen) [inline apex-benchmarks/src/main/java/com/example/JsonGenerator.java]: emit only one tuple in emitTuples() will impact the performance

**Comment by brightchen** [comment]
- the license, should be DT or apache? please consist.
- Code style: I think we'd better follow apache apex code style. such as two space indent etc

File: apex-benchmarks/src/main/java/com/example/Application.java (hunk 4 above)
Comment (sandeshh) [inline apex-benchmarks/src/main/java/com/example/Application.java]: Apache license will be used, even the original code used the same.

File: apex-benchmarks/src/main/java/com/example/DeserializeJSON.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
+/**
+ * Licensed under the terms of the Apache License 2.0. Please see LICENSE file in the project root for terms.
+ */
+package com.example;
```
Comment (vrozov) [inline apex-benchmarks/src/main/java/com/example/DeserializeJSON.java]: Please rename package to apex.benchmark to be consistent with other benchmarks.

File: apex-benchmarks/pom.xml [hunk 7]
```diff
@@ -0,0 +1,371 @@
...
+
+  <!-- change these to the appropriate values -->
+  <name>Apex_Benchmark</name>
+  <description>My Apex Application Description</description>
+
+  <properties>
+    <!-- change this if you desire to use a different version of DataTorrent -->
+    <datatorrent.version>3.4.0</datatorrent.version>
+    <datatorrent.apppackage.classpath>lib/*.jar</datatorrent.apppackage.classpath>
+    <malhar.version>3.4.0-incubating-SNAPSHOT</malhar.version>
```
Comment (tweise) [inline apex-benchmarks/pom.xml]: Should be changed to 3.4.0

File: apex-benchmarks/pom.xml [hunk 8]
```diff
@@ -0,0 +1,371 @@
...
+  <artifactId>apex_benchmark</artifactId>
+  <packaging>jar</packaging>
+
+  <!-- change these to the appropriate values -->
+  <name>Apex_Benchmark</name>
+  <description>My Apex Application Description</description>
+
+  <properties>
+    <!-- change this if you desire to use a different version of DataTorrent -->
+    <datatorrent.version>3.4.0</datatorrent.version>
```
Comment (tweise) [inline apex-benchmarks/pom.xml]: should be apex.version

**Comment by sandeshh** [comment]
@brightchen All the review comments have been addressed. Can you please merge?

File: README.markdown [hunk 9]
```diff
@@ -7,7 +7,7 @@
...
+At Yahoo we have adopted [Apache Storm](https://storm.apache.org) as our stream processing platform of choice.  But that was in 2012 and the landscape has changed significantly singe then. Because of this we really want to know what Storm is good at, where it needs to be improved compared to other systems, and what its limitations are compared to other tools so we can recommend the best tool for the job to our customers.  To do this we started to look for stream processing benchmarks that we could use to do this evaluation, but all of them ended up lacking in several fundamental areas.  Primarily they did not test anything close to a read world use case, so we decided to write a simple one.  This is the first round of these tests.  The tool here is not polished and only covers three tools and one specific use case.  We hope to expand this in the future in terms of the tools tested, the variety of processing tested, and the metrics gathered.
```
Comment (tweise) [inline README.markdown]: Why is it showing a change here?

File: pom.xml [hunk 10]
```diff
@@ -27,6 +27,7 @@ Licensed under the terms of the Apache License 2.0. Please see LICENSE file in t
         <slf4j.version>1.7.7</slf4j.version>
         <commons-cli.version>1.3.1</commons-cli.version>
         <snakeyaml.version>1.11</snakeyaml.version>
+        <apex.version>3.2.0-incubating</apex.version>
```
Comment (tweise) [inline pom.xml]: Looks suspect.

**Comment by sandeshh** [comment]
@tweise Addressed all your comments.

**Pull request merged**